### DISCUSSION TITLE
undefined symbol problem

### DISCUSSION THREAD

**Discussion by WenAce** [emscripten-core/emscripten#19595] [Q&A] [answered]
when i use emcc to build my project,it has a problem in the end,the error is ": undefined symbol: emscripten_dispatch_to_thread_",then I use link command " -s USE_PTHREADS=1 ",and error change “wasm-ld: error: --shared-memory is disallowed by ldo.c.obj because it was not compiled with 'atomics' or 'bulk-memory' features.” so what should i Do next

**Comment by sbc100** [comment, marked as answer]
You need to build all the objects in your project with `-pthread` (or equivalently `-sUSE_PTHREADS`) (in this case it looks like ldo.c.obj in particular was not built with that flag).
